gcloud-java-datastore/src/main/java/com/google/gcloud/spi/DefaultDatastoreRpc.java

```diff
@@ -62,26 +42,27 @@
   }
 
   public DefaultDatastoreRpc(DatastoreOptions options) {
-    client = DatastoreFactory.get().create(
```

Comment:
You can construct the builder once and only keep the condition for host/localhost setting.